File: src/main/java/com/tom/student/Student.java
package com.tom.student;

public class Student {
    String name;
    int english;
    int math;
    static int pass = 60;

    public Student(String name, int english, int math) {
        this.name = name;
        this.english = english;
        this.math = math;
    }

    public int highest() {
        int max = (english>math) ? english : math ;
        /*int max = 0 ;
        if (english>math) {
            max = english;
        }
        else {
            max = math;
        }*/
        return max;
    }

    public void print() {

        int average = getAverage();
        System.out.print(name + "\t" +
                           english + "\t" +
                           math + "\t" +
                           getAverage() + "\t" +
                           ((average>=pass ) ? "PASS" : "FAILED"));

        char grading = 'F';

        switch (average/10) {
            case 10:
            case 9:
                grading = 'A';
                break;
            case 8:
                grading = 'B';
                break;
            case 7:
                grading = 'C';
                break;
            case 6:
                grading = 'D';
                break;
            default:
                grading = 'F';
        }

/*        if (average >= 90 && average <=100){
            grading = 'A';
        } else if (average >=80 && average <= 89){
            grading = 'B';
        } else if (average >=70 && average <= 79){
            grading = 'C';
        } else if (average >=60 && average <=69){
            grading = 'D';
        }*/
        System.out.println("\t"+grading);
 /*       if (getAverage() >= 60) {
            System.out.println("PASS");
        } else {
            System.out.println("FAILED");
        }*/
    }

    public int getAverage() {
        return (english+math)/2;
    }
}

File: src/main/kotlin/com/tom/kotlin/student/student.kt
package com.tom.kotlin.student

import java.util.*

fun main() {
//    userInput()
    student.pass = 50
    val stu = student("xiang",70,100)
    val stu2 = student("kaint",44,68)
    val stu3 = student("reann",60,20)
    val gstu = GraduateStudent("peace",55,65,60)
    gstu.print()
    stu.print()
    stu2.print()
    stu3.print()

    var test = 123
    println("this is : $test")
    println("highest score: ${stu.highest()}")

}

class GraduateStudent (name: String?,english: Int,math: Int,var thesis: Int): student(name, english, math) {
    companion object {
        var pass = 70
    }

    override fun print() {
        println("$name\t$english\t$math\t$thesis\t${getAverage()}\t${passOrFailed()}\t${grading()}")
    }

    override fun passOrFailed(): String = if (getAverage()>= pass) "!PASS!" else "!FAIELD!"

}

// 要加上opent 才可以繼承此方法
open class student(var name:String?, var english:Int, var math:Int){
    companion object {
        @JvmStatic
        var pass = 60;
        fun test() {
            println("testing")
        }
    }

    fun highest() = if (english > math) {
        println("english")
        english
    } else {
        println("math")
        math
    }

    fun getAverage() = (english+math)/2

    //要加上open，才可覆寫此方法
    open fun print(){
        println("$name\t$english\t$math\t${getAverage()}\t${passOrFailed()}\t${grading()}")
    }

    fun grading() = when (getAverage()){
        in 90..100 -> 'A'
        in 80..89 -> 'B'
        in 70..79 -> 'C'
        in 60..69 -> 'D'
        else -> 'F'
    }

    //因graduateStudent中，pass判斷分數不同，因此也要覆寫
    open fun passOrFailed() = if (getAverage()>= pass) "!PASS!" else "!FAIELD!"

    fun nameCheck() = println(name?.length)
}

private fun userInput() {
    val scanner = Scanner(System.`in`) //in在Kotlin為關鍵字，需要``辨別
    print("Please enter student's name : ")
    var name = scanner.next()
    print("Please enter student's english : ")
    var english = scanner.nextInt()
    print("Please enter student's math : ")
    var math = scanner.nextInt()
    var stu = student(name, english, math)
    stu.print()
    stu.nameCheck()
}

